agent/engine/docker_task_engine.go

```diff
@@ -236,6 +236,7 @@ func (engine *DockerTaskEngine) synchronizeState() {
 					log.Warn("Could not find matching container for expected", "name", cont.DockerName)
 				} else {
 					cont.DockerID = describedCont.ID
+					cont.Container.SetKnownStatus(api.ContainerCreated)
```

Comment:
I think we need to look here and set the state based on what comes back.  It could be created, running, or stopped.

Reply:
If it falls into this condition `dockerID==""`, the container status shouldn't be > created. Otherwise the `dockerID` field won't be empty. But have a check here will definitely add some confidence.